### ISSUE TITLE
[Bug]: after upgrading to any version above 0.0.39 three.js ExtrudeGeometry starts to produce wrong results 

### ISSUE THREAD

**Issue by lukasvdelft** [IFCjs/web-ifc#603]
### What happened?

Bases on web-ifc imports I generate Three.js shape objects (2D), which I extrude with native three.js method ExtrudeGeometry by the following line:
const geometry = new ExtrudeGeometry(shape, { depth: distance, bevelEnabled: false });

With web-ifc version 0.0.39 installed it gives me proper objects. When I use any version above 0.0.39 (i tried 4 of them) the extrusion results in >30% percent more vertices with a lot of NaN as values.
How does web-ifc influence Three.js functions? I see in web-ifc version 0.0.41 for instance that nothing changed with packages three and @types/three. 
In the images below (bottom one the correct one), the difference is shown.

<img width="1490" alt="Screenshot 2024-02-06 at 10 36 38" src="https://github.com/IFCjs/web-ifc/assets/103726718/ca548b61-590e-4db2-9591-4031e60d9734">
<img width="1492" alt="Screenshot 2024-02-06 at 10 37 55" src="https://github.com/IFCjs/web-ifc/assets/103726718/d03402e0-326e-4acb-8770-5fded2a3b721">

### Version

0.0.39

### What browsers are you seeing the problem on?

Chrome

**Comment by beachtom** [comment]
Can you share the code you are using? We have no actual dependency on three.js (only for tests and example viewer) so there should be no interaction

**Comment by lukasvdelft** [comment]
yes i agree this is very strange. to make it as clean as possible. first image shows the shape generated based on web-ifc work, this is right, so there is no functional influence of web-ifc anymore from a object perspective (a shape object does not hold refs to the vector2's used to create it). so this shape i feed into a three.js function, i removed all other stuff afterwards (so I return a null object), no influence at all, and the previous send images are the output from the console.log(bufferGeometry) line.
The only thing I do in between is the following:
npm uninstall web-ifc (the 0.0.39 version)
remove the corresponding wasm files
npm install web-ifc@0.0.41
copy the wasm files to the "public" folder (Vue env) and restart the vite/vue setup.

if you want other code, please ask, it's just my assumption that this covers the issue but i might be wrong




![Screenshot 2024-02-06 at 11 17 23](https://github.com/IFCjs/web-ifc/assets/103726718/4d1af777-2eab-470e-ac00-a950ebebdf1f)

<img width="741" alt="Screenshot 2024-02-06 at 11 20 24" src="https://github.com/IFCjs/web-ifc/assets/103726718/4e5495b7-19ef-4ae9-ac3b-86e290c3530c">

**Comment by beachtom** [comment]
How do you get the object from web-ifc? Is it in the StreamAllMeshes function or via a GetLine()?

**Comment by lukasvdelft** [comment]
first GetLineIDsWithType(), then properties.getItemProperties() (recursive) and later the GetRawLineData() for the points.

**Comment by lukasvdelft** [comment]
good idea! 
and I know whats wrong. 0.0.39 gives me ~14 decimals. 0.0.41 gives me 6. And then I end-up with duplicate values.. that explains the NaNs..

**Comment by beachtom** [comment]
So it is a bug in web-ifc? The geometry output may have changed but may still be correct.....

**Comment by lukasvdelft** [comment]
if it is a bug is as subjective matter :) probably a more strict rounding rule has been inserted after 0.0.39. there was probably a good reason for that.. a suggestion could be to make the rounding rule adjustable. I experience it in the case of a stairs with 27 steps as extruded elements, the precision from the Revit output is bigger than web-ifc processes.

**Comment by beachtom** [comment]
so we added extra normalization on all the coordinates - to prevent very large decimals that lead to FP errors in rendering. It should not effect the accuracy of the geometry. Can you validate if that is the case with your geometry?

**Comment by lukasvdelft** [comment]
yes, it is okay. thanks so much Tom, again.

**Comment by beachtom** [comment]
No problem!
